lib/money/money/arithmetic.rb

```diff
@@ -132,10 +132,11 @@ def negative?
         when Money
           other = other.exchange_to(currency)
           new_fractional = fractional.public_send(op, other.fractional)
-          self.class.new(new_fractional, currency, bank)
+          infinite_precision = self.infinite_precision? || other.infinite_precision?
```

Comment:
nit: this a good candidate for a new method with a descriptive name as it isn't immediately obvious that it's favouring a more precise setting

Reply:
I can do that... any naming suggestions?  Something like `arithmetic_precision`?

Reply:
`highest_precision(self, other)`?

Reply:
@Art3606, please work on this one